sdk/core/azure-core/azure/core/tracing/_abstract_span.py

```diff
@@ -129,6 +129,18 @@ def span_instance(self):
         Returns the span the class is wrapping.
         """
 
+    @staticmethod
```

Comment:
Why does a protocol type has a static method? We generally try to avoid `@staticmethod`s, and having it on a protocol seem doubly odd? Especially since it doesn't have any documentation on what it is supposed to return other than a "link"?

Reply:
This method is supposed to create a link given headers so that it can be passed to the child span while creating the span instance. 

the other idea i had is have a custom `Link` class in azure-core exposed along with abstract span.

```
class Link:
  def __init__(self, headers, attributes):
    self.headers = headers
    self.attributes = attributes
```

and ask the users to use this instead of OT's Link
```
from azure.core.tracing import Link
link = Link(headers, attributes)
```
internally, in the OT span __init__, we change this into Opentelemetry's span.k

This felt like an overkill personally
- and right now, this method only exists to create a lin